docs/CONTRIBUTING.md

```diff
@@ -93,6 +93,21 @@ Then run the e2e tests:
 make test-e2e
 ```
 
+## Controller architecture
+
+Argo Rollouts is actually a collection of individual controllers
+that handle a specific aspect of Progressive Delivery.
+
+[![Internal Architecture](architecture-assets/internal-architecture.png)](architecture-assets/internal-architecture.png)
+
+The controllers are:
+
+* [Rollout Controller](https://github.com/argoproj/argo-rollouts/blob/master/controller/controller.go)
```

Comment:
I think the rollout controller code is in `https://github.com/argoproj/argo-rollouts/blob/master/rollout/controller.go`

Reply:
@huikang link was updated.